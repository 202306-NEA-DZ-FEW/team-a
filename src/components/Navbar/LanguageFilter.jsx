export default function LanguageFilter() {
  return (
    <div className='btn btn-circle btn-ghost'>
      <img
        src='/LangIcon.png'
        alt='LangIcon'
        className='object-cover w-full h-full'
      />
    </div>
  );
}
