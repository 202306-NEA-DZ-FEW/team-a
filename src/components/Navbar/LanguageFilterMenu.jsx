import { Menu, Transition } from "@headlessui/react";
import Link from "next/link";
import { Fragment } from "react";
import { TbLanguage } from "react-icons/tb";

function LanguageFilterMenu({ currentPath }) {
  //This is for classnames
  function classNames(...classes) {
    return classes.filter(Boolean).join(" ");
  }
  return (
    <Menu as='div' className='text-left hidden lg:block'>
      <div>
        <Menu.Button className='btn btn-sm outline-dashed transition-all duration-500 ease-in-out hover:outline-double outline-[2px] btn-circle btn-ghost'>
          <TbLanguage className='w-6 h-6' />
        </Menu.Button>
      </div>

      <Transition
        as={Fragment}
        enter='transition ease-out duration-100'
        enterFrom='transform opacity-0 scale-95'
        enterTo='transform opacity-100 scale-100'
        leave='transition ease-in duration-75'
        leaveFrom='transform opacity-100 scale-100'
        leaveTo='transform opacity-0 scale-95'
      >
        <Menu.Items className='absolute right-0 z-40 mt-6 lg:mr-4 w-screen md:w-screen lg:w-56 rounded-3xl p-2 bg-white shadow-xl ring-1 ring-black ring-opacity-5 focus:outline-none font-light tracking-wider'>
          <Menu.Item>
            {({ active }) => (
              <Link
                href={currentPath}
                locale='en'
                className={classNames(
                  active ? "bg-gray-200 text-gray-900" : "text-gray-700",
                  "block px-4 py-2 text-sm my-2 rounded-md"
                )}
              >
                <span className='fi fi-gb rounded mr-2'></span> English
              </Link>
            )}
          </Menu.Item>
          <Menu.Item>
            {({ active }) => (
              <Link
                href={currentPath}
                locale='fr'
                className={classNames(
                  active ? "bg-gray-200 text-gray-900" : "text-gray-700",
                  "block px-4 py-2 text-sm mb-2 rounded-md"
                )}
              >
                <span className='fi fi-fr rounded mr-2'></span> Français
              </Link>
            )}
          </Menu.Item>
          <Menu.Item>
            {({ active }) => (
              <Link
                href={currentPath}
                locale='tr'
                className={classNames(
                  active ? "bg-gray-200 text-gray-900" : "text-gray-700",
                  "block px-4 py-2 text-sm mb-2 rounded-md"
                )}
              >
                <span className='fi fi-tr rounded mr-2'></span> Türkçe
              </Link>
            )}
          </Menu.Item>
          <Menu.Item>
            {({ active }) => (
              <Link
                href={currentPath}
                locale='ar'
                className={classNames(
                  active ? "bg-gray-200 text-gray-900" : "text-gray-700",
                  "block px-4 py-2 text-sm mb-2 rounded-md"
                )}
              >
                <span className='fi fi-dz rounded mr-2'></span> العربية
              </Link>
            )}
          </Menu.Item>
        </Menu.Items>
      </Transition>
    </Menu>
  );
}

export default LanguageFilterMenu;
