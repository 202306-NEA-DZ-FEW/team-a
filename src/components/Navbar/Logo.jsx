import Image from "next/image";

export default function Logo() {
  return (
    <div>
      <img src='/Logo.png' className='h-14 w-28 object-cover' />
    </div>
  );
}
