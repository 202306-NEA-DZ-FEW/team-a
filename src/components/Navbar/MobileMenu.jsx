import Link from "next/link";
import { useRouter } from "next/router";
import React, { useEffect, useState } from "react";
import { BiLogOut } from "react-icons/bi";
import { HiOutlineMenuAlt3, HiOutlineX } from "react-icons/hi";

function MobileMenu({ user, t, logOut }) {
  const [open, setopen] = useState(false);
  const [currentPath, setCurrentPath] = useState("/");
  const router = useRouter();
  useEffect(() => {
    // Get the current path
    const currentPath = router.asPath;
    setCurrentPath(currentPath);
  }, [router]);

  const handleClick = () => {
    setopen(false);
  };
  return (
    <nav className='lg:hidden no-scrollbar navbar-end z-50 no-scrollbar'>
      <button
        onClick={() => setopen(!open)}
        className='z-50 justify-center px-3 py-2 text-sm font-semibold btn-circle btn-ghost'
      >
        {open ? (
          <HiOutlineX className={`${open ? "text-white" : ""} w-6 h-6`} />
        ) : (
          <HiOutlineMenuAlt3
            className={`${open ? "text-white" : ""} w-6 h-6`}
          />
        )}
      </button>
      <div
        className={`${
          open ? "flex" : "hidden"
        } flex-col z-40 fixed justify-center w-screen h-screen bg-neutral top-0 left-0`}
      >
        <div className='w-full flex justify-center gap-4 items-center mt-10 '>
          <Link
            onClick={handleClick}
            href={currentPath}
            locale='en'
            className='font-light p-2 transition-all duration-500 hover:-translate-y-1 text-xl text-white'
          >
            <span className='fi fi-gb rounded text-4xl'></span>
          </Link>
          <Link
            onClick={handleClick}
            href={currentPath}
            locale='fr'
            className='font-light p-2 transition-all duration-500 hover:-translate-y-1 text-xl text-white'
          >
            <span className='fi fi-fr rounded text-4xl'></span>
          </Link>
          <Link
            onClick={handleClick}
            href={currentPath}
            locale='tr'
            className='font-light p-2 transition-all duration-500 hover:-translate-y-1 text-xl text-white'
          >
            <span className='fi fi-tr rounded text-4xl'></span>
          </Link>
          <Link
            onClick={handleClick}
            href={currentPath}
            locale='ar'
            className='font-light p-2 transition-all duration-500 hover:-translate-y-1 text-xl text-white'
          >
            <span className='fi fi-dz rounded text-4xl'></span>
          </Link>
        </div>
        {user && (
          <Link
            onClick={handleClick}
            className='py-8 text-center transition-all duration-500 hover:translate-x-1 normal-case font-light text-5xl text-white tracking-wider'
            href='/'
          >
            {t("common:navbar:dashboard")}
          </Link>
        )}
        <Link
          onClick={handleClick}
          className='py-8 text-center transition-all duration-500 hover:translate-x-1 normal-case font-light text-5xl text-white tracking-wider'
          href='/'
        >
          {t("common:buttons:home")}
        </Link>
        <Link
          onClick={handleClick}
          className='py-8 text-center transition-all duration-500 hover:translate-x-1 normal-case font-light text-5xl text-white tracking-wider'
          href='/about'
        >
          {t("common:buttons:about")}
        </Link>
        <Link
          onClick={handleClick}
          className='py-8 text-center transition-all duration-500 hover:translate-x-1 normal-case font-light text-5xl text-white tracking-wider'
          href='/items'
        >
          {t("common:buttons:items")}
        </Link>
        <Link
          onClick={handleClick}
          className='py-8 text-center transition-all duration-500 hover:translate-x-1 normal-case font-light text-5xl text-white tracking-wider'
          href='/blogs'
        >
          {t("common:buttons:blogs")}
        </Link>
        {!user && (
          <Link
            href='/auth/sign-in'
            onClick={handleClick}
            className='btn mt-10 btn-active rounded-3xl normal-case font-normal tracking-wider w-[80%] self-center'
          >
            Login
          </Link>
        )}
        {user && (
          <button
            onClick={logOut}
            className='btn text-white text-xl normal-case font-light btn-ghost tracking-wider w-[80%] self-center'
          >
            <BiLogOut size={30} />
            Logout
          </button>
        )}
      </div>
    </nav>
  );
}

export default MobileMenu;
