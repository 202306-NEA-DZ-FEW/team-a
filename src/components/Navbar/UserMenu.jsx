import { Menu, Transition } from "@headlessui/react";
import Image from "next/image";
import Link from "next/link";
import { Fragment } from "react";
import { BiLogOut } from "react-icons/bi";
import { MdOutlineDashboardCustomize } from "react-icons/md";

function UserMenu({ user, logOut, t, userProfile }) {
  function classNames(...classes) {
    return classes.filter(Boolean).join(" ");
  }

  return (
    <Menu as='div' className='relative hidden lg:block'>
      <Menu.Button className='btn btn-sm outline-dashed transition-all duration-500 ease-in-out hover:outline-double outline-[2px] btn-circle btn-ghost'>
        <div className='avatar'>
          <div className='w-8 rounded-full ring ring-primary hover:ring-offset-1 transition-all duration-500 ring-offset-base-100 ring-offset-2'>
            <Image width={100} height={100} alt='user' src={userProfile} />
          </div>
        </div>
      </Menu.Button>
      <Transition
        as={Fragment}
        enter='transition ease-out duration-100'
        enterFrom='transform opacity-0 scale-95'
        enterTo='transform opacity-100 scale-100'
        leave='transition ease-in duration-75'
        leaveFrom='transform opacity-100 scale-100'
        leaveTo='transform opacity-0 scale-95'
      >
        <Menu.Items className='absolute right-0 translate-x-1/3 left-auto origin-center z-40 top-12 lg:mr-4 w-screen md:w-screen lg:w-56 rounded-xl p-2 bg-base-100 shadow-xl ring-1 ring-black ring-opacity-5 focus:outline-none font-light tracking-wider'>
          <Menu.Item>
            {({ active }) => (
              <Link
                className={classNames(
                  active ? "bg-base-300" : "text-gray-700",
                  "px-4 py-2 text-sm mb-2 rounded-md w-full flex gap-1 items-center"
                )}
                href={{
                  pathname: "/dashboard",
                  query: { user: user.uid },
                }}
              >
                <MdOutlineDashboardCustomize size={20} />
                {t("common:navbar:dashboard")}
              </Link>
            )}
          </Menu.Item>
          <Menu.Item>
            {({ active }) => (
              <button
                className={classNames(
                  active ? "bg-base-300" : "text-gray-700",
                  "px-4 py-2 text-sm mb-2 rounded-md w-full flex gap-1 items-center"
                )}
                onClick={logOut}
              >
                <BiLogOut size={20} />
                {t("common:buttons:signOut")}
              </button>
            )}
          </Menu.Item>
        </Menu.Items>
      </Transition>
    </Menu>
  );
}

export default UserMenu;
