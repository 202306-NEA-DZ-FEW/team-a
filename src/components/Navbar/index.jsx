import Link from "next/link";
import LanguageFilterMenu from "./LanguageFilterMenu";
import UserMenu from "./UserMenu";
import MobileMenu from "./MobileMenu";

function Navbar() {
  const user = { id: 1, name: "Farouk the Legend", imageUrl: "url" };
  return (
    <>
      {/* Navbar Starts Here */}
      <div className='navbar bg-white'>
        <div className='navbar-start ml-7'>
          {/* Navbar Logo */}
          <Link href={"/"}>
            <p className='font-bold text-2xl text-black'>Unify.</p>
          </Link>
        </div>
        <div className='navbar-center hidden lg:block'>
          {/* Navbar Navigation Links */}
          <ul className='menu menu-horizontal px-1 mx-1 text-black'>
            <li>
              <Link className='hover:bg-cyan-800 hover:text-white' href={"/"}>
                Home
              </Link>
            </li>
            <li>
              <Link
                className='hover:bg-cyan-800 hover:text-white'
                href={"/about"}
              >
                About Us
              </Link>
            </li>
            <li>
              <Link
                className='hover:bg-cyan-800 hover:text-white'
                href={"/products"}
              >
                Products
              </Link>
            </li>
            <li>
              <Link
                className='hover:bg-cyan-800 hover:text-white'
                href={"/blogs"}
              >
                Blogs
              </Link>
            </li>
          </ul>
        </div>
        <div className='navbar-end mr-7'>
          {/* Language Filter */}
          <LanguageFilterMenu />
          {/* Avatar/Sign-in Button */}
          <UserMenu user={user} />
          {/* This is Related to Mobile View Navbar Menu */}
          <MobileMenu user={user} />
        </div>
      </div>
      {/* Navbar Ends Here */}
    </>
  );
}

export default Navbar;
