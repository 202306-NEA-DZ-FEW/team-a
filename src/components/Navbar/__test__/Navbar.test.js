import renderer from "react-test-renderer";
import Navbar from "..";

it("renders correctly", () => {
  const tree = renderer.create(<Navbar />).toJSON();
  expect(tree).toMatchSnapshot();
});
