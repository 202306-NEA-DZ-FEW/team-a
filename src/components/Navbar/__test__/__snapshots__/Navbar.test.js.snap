// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar snapshot test 1`] = `
<header
  className="no-scrollbar backdrop-blur-sm bg-white bg-opacity-40 sticky top-0 left-0 navbar gap-24 justify-between 2xl:px-32 xl:px-28 px-10 z-50"
>
  <div
    className="lg:block navbar-start max-w-fit"
  >
    <a
      className="font-bold btn btn-ghost normal-case text-2xl"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      buttons.logo
    </a>
  </div>
  <nav
    className="hidden w-full flex-1 lg:flex lg:navbar-end gap-4 lg:items-center"
  >
    <div
      className="w-1/6 h-[2.2px] bg-black"
    />
    <div
      className="lg:flex gap-4"
      dir="ltr"
    >
      <a
        className="btn hover:btn-link transition-all duration-500 hover:translate-y-1 btn-ghost normal-case font-light tracking-wider"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        buttons.home
      </a>
      <a
        className="btn hover:btn-link transition-all duration-500 hover:translate-y-1 btn-ghost normal-case font-light tracking-wider"
        href="/about"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        buttons.about
      </a>
      <a
        className="btn hover:btn-link transition-all duration-500 hover:translate-y-1 btn-ghost normal-case font-light tracking-wider"
        href="/items"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        buttons.items
      </a>
      <a
        className="btn hover:btn-link transition-all duration-500 hover:translate-y-1 btn-ghost normal-case font-light tracking-wider"
        href="/blogs"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        buttons.blogs
      </a>
    </div>
    <div
      className="text-left hidden lg:block"
      data-headlessui-state=""
    >
      <div>
        <button
          aria-expanded={false}
          aria-haspopup="menu"
          className="btn btn-sm outline-dashed transition-all duration-500 ease-in-out hover:outline-double outline-[2px] btn-circle btn-ghost"
          data-headlessui-state=""
          id="headlessui-menu-button-:r0:"
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          type="button"
        >
          <svg
            className="w-6 h-6"
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
              stroke="none"
            />
            <path
              d="M4 5h7"
            />
            <path
              d="M9 3v2c0 4.418 -2.239 8 -5 8"
            />
            <path
              d="M5 9c0 2.144 2.952 3.908 6.7 4"
            />
            <path
              d="M12 20l4 -9l4 9"
            />
            <path
              d="M19.1 18h-6.2"
            />
          </svg>
        </button>
      </div>
    </div>
  </nav>
  <nav
    className="lg:hidden no-scrollbar navbar-end z-50 no-scrollbar"
  >
    <button
      className="z-50 justify-center px-3 py-2 text-sm font-semibold btn-circle btn-ghost"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className=" w-6 h-6"
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 6h16M4 12h16m-7 6h7"
          strokeLinecap="round"
          strokeLinejoin="round"
        />
      </svg>
    </button>
    <div
      className="hidden flex-col z-40 fixed justify-center w-screen h-screen bg-neutral top-0 left-0"
    >
      <div
        className="w-full flex justify-center gap-4 items-center mt-10 "
      >
        <a
          className="font-light p-2 transition-all duration-500 hover:-translate-y-1 text-xl text-white"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <span
            className="fi fi-gb rounded text-4xl"
          />
        </a>
        <a
          className="font-light p-2 transition-all duration-500 hover:-translate-y-1 text-xl text-white"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <span
            className="fi fi-fr rounded text-4xl"
          />
        </a>
        <a
          className="font-light p-2 transition-all duration-500 hover:-translate-y-1 text-xl text-white"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <span
            className="fi fi-tr rounded text-4xl"
          />
        </a>
        <a
          className="font-light p-2 transition-all duration-500 hover:-translate-y-1 text-xl text-white"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <span
            className="fi fi-dz rounded text-4xl"
          />
        </a>
      </div>
      <a
        className="py-8 text-center transition-all duration-500 hover:translate-x-1 normal-case font-light text-5xl text-white tracking-wider"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        buttons.home
      </a>
      <a
        className="py-8 text-center transition-all duration-500 hover:translate-x-1 normal-case font-light text-5xl text-white tracking-wider"
        href="/about"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        buttons.about
      </a>
      <a
        className="py-8 text-center transition-all duration-500 hover:translate-x-1 normal-case font-light text-5xl text-white tracking-wider"
        href="/items"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        buttons.items
      </a>
      <a
        className="py-8 text-center transition-all duration-500 hover:translate-x-1 normal-case font-light text-5xl text-white tracking-wider"
        href="/blogs"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        buttons.blogs
      </a>
      <a
        className="btn mt-10 btn-active rounded-3xl normal-case font-normal tracking-wider w-[80%] self-center"
        href="/auth/sign-in"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Login
      </a>
    </div>
  </nav>
</header>
`;
